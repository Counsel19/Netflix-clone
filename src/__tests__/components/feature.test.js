import React from "react";
import { render } from "@testing-library/react";
import { Feature, OptForm } from "../../components";

describe("<Feature />", () => {
  it("renders the <Feature /> component with populated data", () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>Unlimited Films, Tv Programs and more.</Feature.Title>
        <Feature.SubTitle>Watch anywhere, Cancel at anytime</Feature.SubTitle>

        <OptForm>
          <OptForm.Input placeholder="Email Adress" />
          <OptForm.Button>Try it Now</OptForm.Button>

          <OptForm.Break />

          <OptForm.Text>
            Ready to watch? Enter your email to create or restart your
            membership
          </OptForm.Text>
        </OptForm>
      </Feature>
    );

    expect(getByText('Unlimited Films, Tv Programs and more.')).toBeTruthy();
    expect(getByText('Watch anywhere, Cancel at anytime')).toBeTruthy();
    expect(getByText('Ready to watch? Enter your email to create or restart your membership')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot()
  });

  it("renders the <Feature /> component with only a title", () => {
    const { container, getByText, queryByText } = render(
      <Feature>
        <Feature.Title>Unlimited Films, Tv Programs and more.</Feature.Title>
      </Feature>
    );

    expect(getByText('Unlimited Films, Tv Programs and more.')).toBeTruthy();
    expect(queryByText('Watch anywhere, Cancel at anytime')).toBeFalsy();
    expect(queryByText('Ready to watch? Enter your email to create or restart your membership')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot()

  });

});
