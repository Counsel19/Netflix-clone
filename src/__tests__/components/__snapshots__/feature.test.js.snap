// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders the <Feature /> component with only a title 1`] = `
<div
  class="feature__Container-sc-tsydtl-0 eEKEHo"
>
  <h1
    class="feature__Title-sc-tsydtl-1 bcQSWe"
  >
    Unlimited Films, Tv Programs and more.
  </h1>
</div>
`;

exports[`<Feature /> renders the <Feature /> component with populated data 1`] = `
<div
  class="feature__Container-sc-tsydtl-0 eEKEHo"
>
  <h1
    class="feature__Title-sc-tsydtl-1 bcQSWe"
  >
    Unlimited Films, Tv Programs and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-tsydtl-2 dwixDm"
  >
    Watch anywhere, Cancel at anytime
  </h2>
  <div
    class="opt-form__Container-sc-q4a5n7-0 itoax"
  >
    <input
      class="opt-form__Input-sc-q4a5n7-1 kGRwEn"
      placeholder="Email Adress"
    />
    <button
      class="opt-form__Button-sc-q4a5n7-2 laQfqN"
    >
      Try it Now
      <img
        alt="icon"
        src="./images/icons/chevron-right.png"
      />
    </button>
    <div
      class="opt-form__Break-sc-q4a5n7-4 eCanAT"
    />
    <p
      class="opt-form__Text-sc-q4a5n7-3 eYIOjO"
    >
      Ready to watch? Enter your email to create or restart your membership
    </p>
  </div>
</div>
`;
